src/registry.rs

```diff
@@ -69,20 +69,23 @@ impl RegistryCore {
         let mut mf_by_name = HashMap::new();
 
         for c in self.colloctors_by_id.values() {
-            let mf = c.collect();
+            let mut mf = c.collect();
             let name = mf.get_name().to_owned();
-            if let Vacant(entry) = mf_by_name.entry(name.clone()) {
-                entry.insert(mf);
-                continue;
-            }
-
-            let mut existent_mf = mf_by_name.get_mut(&name).unwrap();
-            let mut existent_metrics = existent_mf.mut_metric();
 
-            // TODO: check type.
-            // TODO: check consistency.
-            for metric in mf.get_metric() {
-                existent_metrics.push(metric.clone())
+            match mf_by_name.entry(name) {
+                Entry::Vacant(entry) => {
+                    entry.insert(mf);
+                }
+                Entry::Occupied(mut entry) => {
+                    let mut existent_mf = entry.get_mut();
+                    let mut existent_metrics = existent_mf.mut_metric();
+
+                    // TODO: check type.
+                    // TODO: check consistency.
+                    for metric in mf.take_metric().into_iter() {
```

Comment:
`existent_metrics` is a [`protobuf::RepeatedField`](https://github.com/stepancheg/rust-protobuf/blob/master/src/lib/repeated.rs), unfortunately it does not support extend.